dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/generator/CustomTagGenerator.java

```diff
@@ -560,45 +492,51 @@ private void assembleTagFromWriters(JavaCodeWriter writer) throws JspCoreExcepti
         if (tagGenerator.fragmentWriterUsed() == false) {
             String body = tagGenerator.getBodyWriter().toString(); //232818
             if (body.length() > 0) { //232818
-            	writeDebugStartEnd(tagWriter);
+                writeDebugStartEnd(tagWriter);
                 debugEnd = true; //232818
             }
             tagWriter.printMultiLn(body); //232818
             if (debugEnd) //232818
-            	writeDebugEndBegin(tagWriter);
+                writeDebugEndBegin(tagWriter);
         }
         tagWriter.printMultiLn(tagEndWriter.toString());
 
         if (genTagInMethod) {
             if (methodNesting > 0) {
                 methodWriter.println("return false;");
+
+                if ((reuseTag == false) && !(tagGenerator instanceof SimpleTagGenerator)) {
+                    methodWriter.println("} finally {");
+                    methodWriter.println("_jsp_cleanUpTag(" + tagHandlerVar + ", null);");
+                    methodWriter.println("}");
+                }
```

Comment:
Previously, the clean up code (i.e doPreDestroy) was in the generateEndTag methods. I've placed them it here in this finally block again.

This only applies if we are within a tag method.